apps/test-app/frontend/src/components/tree-widget/StatelessTreeWidget.tsx

```diff
@@ -46,49 +46,48 @@ export function StatelessTreeWidget(props: Omit<TreeWidgetProps, "rulesetId">) {
     setMetadataProvider(createMetadataProvider(schemas));
   }, [props.imodel]);
 
-  const { value } = useDebouncedAsyncValue(
+  const { value: filteredPaths } = useDebouncedAsyncValue(
     useCallback(async () => {
-      if (metadataProvider && queryExecutor && filter !== "") {
-        setFilteringStatus(FilteringInputStatus.FilteringInProgress);
+      if (!metadataProvider || !queryExecutor) {
+        return undefined;
+      }
+      if (filter !== "") {
+        setFilteringStatus("filtering");
```

Comment:
Not sure... At the moment I have it set up so the filtering status is reset to `ready` from the `onItemsRendered` callback, because filtering is being done in 2 stages: getting the paths and filtering using them. Using `inProgress` would only cover the first stage, but not the second one.